src/lib.rs

```diff
@@ -21,17 +21,24 @@ use pyo3::prelude::*;
 
 // Re-export Apache Arrow DataFusion dependencies
 pub use datafusion;
-pub use datafusion_common;
-pub use datafusion_expr;
-pub use datafusion_optimizer;
-pub use datafusion_sql;
+pub mod datafusion_expr {
+    pub use datafusion::logical_expr::*;
+}
+pub use datafusion::optimizer;
+pub mod datafusion_sql {
+    pub use datafusion::sql::*;
+}
+pub mod datafusion_common {
+    pub use datafusion::common::*;
+}
```

Comment:
It might be a little cleaner as:

```
pub use datafusion::logical_expr as datafusion_expr;
pub use datafusion::optimizer;
pub use datafusion::sql as datafusion_sql;
pub use datafusion::common as datafusion_common;
```

Reply:
done!